prov/gni/include/gnix_cm_nic.h

```diff
@@ -66,7 +66,7 @@ typedef int gnix_cm_nic_rcv_cb_func(struct gnix_cm_nic *cm_nic,
 struct gnix_cm_nic {
 	struct gnix_nic *nic;
 	struct gnix_dgram_hndl *dgram_hndl;
-	struct gnix_fid_fabric *fabric;
+	struct gnix_fid_domain *domain;
```

Comment:
The domain has a pointer to the cm_nic which needs to be NULLed during destruct of the cm_nic. We therefore, need a pointer back to the domain in the cm_nic to NULL. It seemed redundant to have the domain and fabric pointer in the cm_nic struct.